### ISSUE TITLE
1.0.0 release

### ISSUE THREAD

**Issue by ko1** [ruby/debug#148]
I'm planning to release 1.0.0.

Any other TODOs?

**Comment by ko1** [comment]
`fork` support is one big topic, but 1.1.0 is also fine IMO.

**Comment by st0012** [comment]
Can we add a way to disable warnings (and perhaps a default)? I set log debug level in dev but I don't want to see these debugger logs every time I use the debugger.

```
DEBUGGER: Load /Users/st0012/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/railties-6.1.4/lib/rails/application/bootstrap.rb
DEBUGGER: Load /Users/st0012/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/railties-6.1.4/lib/rails/application/finisher.rb
DEBUGGER: Load /Users/st0012/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/activesupport-6.1.4/lib/active_support/all.rb
DEBUGGER: Load /Users/st0012/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/activesupport-6.1.4/lib/active_support/core_ext.rb
DEBUGGER: Load /Users/st0012/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/activesupport-6.1.4/lib/active_support/core_ext/array.rb
DEBUGGER: Load /Users/st0012/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/activesupport-6.1.4/lib/active_support/core_ext/array/grouping.rb
DEBUGGER: Load /Users/st0012/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/activesupport-6.1.4/lib/active_support/core_ext/array/inquiry.rb
DEBUGGER: Load /Users/st0012/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/activesupport-6.1.4/lib/active_support/array_inquirer.rb
DEBUGGER: Load /Users/st0012/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/activesupport-6.1.4/lib/active_support/core_ext/big_decimal.rb
DEBUGGER: Load /Users/st0012/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/activesupport-6.1.4/lib/active_support/core_ext/digest.rb
DEBUGGER: Load /Users/st0012/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/activesupport-6.1.4/lib/active_support/core_ext/file.rb
DEBUGGER: Load /Users/st0012/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/activesupport-6.1.4/lib/active_support/core_ext/hash.rb
DEBUGGER: Load /Users/st0012/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/activesupport-6.1.4/lib/active_support/core_ext/hash/deep_transform_values.rb
DEBUGGER: Load /Users/st0012/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/activesupport-6.1.4/lib/active_support/core_ext/integer.rb
DEBUGGER: Load /Users/st0012/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/activesupport-6.1.4/lib/active_support/core_ext/integer/multiple.rb
DEBUGGER: Load /Users/st0012/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/activesupport-6.1.4/lib/active_support/core_ext/integer/inflections.rb
DEBUGGER: Load /Users/st0012/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/activesupport-6.1.4/lib/active_support/core_ext/kernel.rb
DEBUGGER: Load /Users/st0012/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/activesupport-6.1.4/lib/active_support/core_ext/kernel/concern.rb
DEBUGGER: Load /Users/st0012/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/activesupport-6.1.4/lib/active_support/core_ext/kernel/singleton_class.rb
DEBUGGER: Load /Users/st0012/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/activesupport-6.1.4/lib/active_support/core_ext/marshal.rb
```

**Comment by st0012** [comment]
Maybe we can colorize `pp` as well? I can make a PR for it quickly.
How about let's set the deadline to this Sunday. All features must be submitted before it, then you can prepare and make the release next week?

**Comment by ko1** [comment]
so we need to define log level...

**Comment by ko1** [comment]
Maybe 2aacf6ce055f0aae2856391c24a95af573df3b42 solves the logging issue.
Please file another issue and comment it as release stopper.

**Comment by st0012** [comment]
> Maybe 2aacf6c solves the logging issue.

Thanks for this. I think it works well now 👍 

> Please file another issue and comment it as release stopper.

Sorry, which issue do you mean? The colorizing `pp` command one has been merged.

**Comment by ko1** [comment]
> Sorry, which issue do you mean? The colorizing pp command one has been merged.

"If you have other issues" I want to say.

**Comment by st0012** [comment]
Some thoughts after using `debug` for entire day on a work project (large Rails app):

- Most useful commands to me:
  - `binding.bp(command: "catch SomeException ;; c")`
  - `binding.bp(command: "info")` (I want this to stop)
  - `bt /pattern/`
  - `frame num`
  - `info`

- Things to improve (imo):
  - Disable `C-c` activation by default - I use RSpec with debugger enabled (`gem "debug"`). And when I want to cancel the build with Ctrl-c, it enters debug session instead. I personally don't want this binding to be activated by default.
  - Pressing `<up>` for the previous command doesn't work with my work project setup. I need to press like 5~8 times to show it. But I don't have such issue when using `exe/rdbg` directly. I'll investigate this more.
  - I often use `p` for evaluation instead of `e[val]` because I want to see the return value as well. I know sometimes we want to ignore the return value like doing `do_operation;` in `irb`. But it'd be odd if people eventually use `p` for evaluation more than `eval` 🤔 


But generally speaking, it's an awesome tool. I don't think anyone would confuse it with `pry` because it can do so much more than it.

**Comment by ko1** [comment]
> Disable C-c activation by default - I use RSpec with debugger enabled (gem "debug"). And when I want to cancel the build with Ctrl-c, it enters debug session instead. I personally don't want this binding to be activated by default.

debug.rb doesn't activate this C-c feature, and it was my bug.

**Comment by ko1** [comment]
> I often use p for evaluation instead of e[val] because I want to see the return value as well. I know sometimes we want to ignore the return value like doing do_operation; in irb. But it'd be odd if people eventually use p for evaluation more than eval 🤔

I can't catch the points. Usually I only use `p` to call any functions on gdb, so I also think only `p` is used on this debugger.

**Comment by st0012** [comment]
oh I thought `e` is designed for normal evaluation and `p` is designed for printing only. I guess we can ignore this point then 😂

**Comment by st0012** [comment]
The current version of debugger can cover most of my daily debug use cases. So I think it's ready for version 1.0 once the `outline` command is merged.

**Comment by ko1** [comment]
Now I'm waiting for reline version up to support multi-threading.
I'll skip reline support if it takes more time.

Using reline, we can change the prompt or color with input text, so we can recognize it is debug command or not. It means there is no confusion about Ruby expression and debug command, so we can allow to accept Ruby expression without `p` like REPL.

**Comment by st0012** [comment]
Wow that'll be a big plus for the debugger 😍

**Comment by st0012** [comment]
@ko1 with the new reline integration, I think it's ready for the `beta9` release?

**Comment by ko1** [comment]
released.
